File: backend/src/main/java/com/delivery/deliverySlot/entity/DeliverySlot.java
package com.delivery.deliverySlot.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "delivery_slots")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class DeliverySlot {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private Integer capacity;

    @Enumerated(EnumType.STRING)
    private DeliverySlotStatus status;
}

File: frontend/src/interfaces/DeliverySlot.ts
export interface DeliverySlot {
    technicalId: number;
    id: number;
    startTime: Date,
    endTime: Date,
    capacity: number,
    status: DeliverySlotStatus
}

export enum DeliverySlotStatus {
    OPEN,
    PENDING,
    CLOSE
}